addons/mail/views/mail_activity_views.xml

```diff
@@ -223,19 +223,27 @@
         <field name="model">mail.activity</field>
         <field name="arch" type="xml">
             <search string="Activity">
+                <field name="res_name"/>
                 <field name="res_model"/>
                 <field name="summary"/>
                 <field name="activity_type_id"/>
-                <filter invisible="1" string="Late Activities" name="activities_overdue"
+                <filter name="activities_my" string="My Activities" domain="[('user_id', '=', uid)]"/>
+                <separator/>
+                <filter string="Overdue" name="activities_overdue"
                         domain="[('date_deadline', '&lt;', context_today().strftime('%Y-%m-%d'))]"
                         help="Show all records which has next action date is before today"/>
-                <filter invisible="1" string="Today Activities" name="activities_today"
+                <filter string="Today" name="activities_today"
                         domain="[('date_deadline', '=', context_today().strftime('%Y-%m-%d'))]"/>
-                <filter invisible="1" string="Future Activities" name="activities_upcoming_all"
+                <filter string="Future" name="activities_upcoming_all"
                         domain="[('date_deadline', '&gt;', context_today().strftime('%Y-%m-%d'))
                         ]"/>
+                <separator/>
+                <filter name="activities_done" string="Done" domain="[('active', '=', False)]"/>
+                <separator />
                 <group expand="0" string="Group By">
                     <filter string="Deadline" name="date_deadline" context="{'group_by': 'date_deadline'}"/>
+                    <filter string="Document Model" name="res_model" context="{'group_by': 'res_model_id'}"/>
```

Comment:
I bet this crashes with non admin people

Reply:
it does not

Reply:
Wait, what ?